lib/client/rpc/market-clients/ammMarkets.ts

```diff
@@ -78,82 +74,106 @@ export class FutarchyAmmMarketsRPCClient implements FutarchyAmmMarketsClient {
+    ammMarket: AmmMarket,
+    quoteAmount: number,
+    maxBasAmount: number
+  ): LiquidityAddError | null {
+    const quoteAmountArg = new BN(
+      quoteAmount *
+        new BN(10).pow(new BN(ammMarket.quoteToken.decimals)).toNumber()
+    );
+    const baseAmountArg = new BN(
+      maxBasAmount *
+        new BN(10).pow(new BN(ammMarket.baseToken.decimals)).toNumber()
+    );
+    // base passed in should be ammBaseAmount honestly...
+    const baseReserve = new BN(ammMarket.baseAmount);
+    const quoteReserve = new BN(ammMarket.quoteAmount);
+
+    const ammBaseAmount = quoteAmountArg
+      .mul(baseReserve)
+      .div(quoteReserve)
+      .add(new BN(1));
+    if (baseAmountArg.toNumber() < ammBaseAmount.toNumber()) {
+      console.error(
+        `liquidity max base exceeded. baseAmountArg: ${baseAmountArg.toNumber()}. quoteAmountArg: ${quoteAmountArg.toNumber()}, ammBaseAmount: ${ammBaseAmount.toNumber()}`
+      );
+      return "AddLiquidityMaxBaseExceeded";
+    }
+    return null;
+  }
+
   async addLiquidity(
-    ammAddr: PublicKey,
-    quoteAmount?: number,
-    baseAmount?: number
-  ) {
+    ammMarket: AmmMarket,
+    quoteAmount: number,
+    maxBasAmount: number
+  ): Promise<string[] | LiquidityAddError> {
     if (!this.transactionSender) return [];
 
-    const ammAcount = await this.ammClient.getAmm(ammAddr);
-    const minLpTokens = new BN(0);
-
-    const quoteInfo = await this.rpcProvider.connection.getAccountInfo(
-      ammAcount.quoteMint
-    );
-    const quoteDecimals = unpackMint(ammAcount.quoteMint, quoteInfo).decimals;
-    const quoteAmountCasted = new BN(quoteAmount).mul(
-      new BN(10).pow(new BN(quoteDecimals))
+    const validationError = this.validateAddLiquidity(
+      ammMarket,
+      quoteAmount,
+      maxBasAmount
     );
+    if (validationError) {
+      return validationError;
+    }
 
-    const baseInfo = await this.rpcProvider.connection.getAccountInfo(
-      ammAcount.baseMint
+    const quoteAmountArg = new BN(
+      quoteAmount *
+        new BN(10).pow(new BN(ammMarket.quoteToken.decimals)).toNumber()
     );
-    const baseDecimals = unpackMint(ammAcount.quoteMint, baseInfo);
-    const baseAmountCasted = new BN(baseAmount).mul(
-      new BN(10).pow(new BN(baseDecimals))
+
+    // we just pass this in as min LP tokens and this replicates the calculation in the program
+    const lpTokensToMint = quoteAmountArg
+      .mul(new BN(ammMarket.lpMintSupply))
+      .div(ammMarket.quoteAmount);
+
+    const baseAmountArg = new BN(
+      maxBasAmount *
+        new BN(10).pow(new BN(ammMarket.baseToken.decimals)).toNumber()
     );
 
     const ix = this.ammClient.addLiquidityIx(
-      ammAddr,
-      ammAcount.baseMint,
-      ammAcount.quoteMint,
-      baseAmountCasted,
-      quoteAmountCasted,
-      minLpTokens,
+      ammMarket.publicKey,
+      ammMarket.baseMint,
+      ammMarket.quoteMint,
+      quoteAmountArg,
+      baseAmountArg,
+      lpTokensToMint,
```

Comment:
This implementation is a misnomer, we need to use the actual values in here.

https://github.com/metaDAOproject/futarchy/blob/3b38e539f1fe50fef77c234961be8b3acaead7ce/futarchy-ts/src/AmmClient.ts#L237C5-L239C21

maxBaseAmount needs to be **multiplied** by  `(1 - slippage)`
minLpTokens needs to be  needs to be **multiplied** by `(1 - slippage)`

Reply:
Names have been changed, but I will do slippage math on frontend

Reply:
psyche, I will do slippage math in the SDK